numpy/lib/arraysetops.py

```diff
@@ -270,20 +270,24 @@ def unique(ar, return_index=False, return_inverse=False,
 
     # Must reshape to a contiguous 2D array for this to work...
     orig_shape, orig_dtype = ar.shape, ar.dtype
-    ar = ar.reshape(orig_shape[0], -1)
+    ar = ar.reshape(orig_shape[0], np.prod(orig_shape[1:]))
     ar = np.ascontiguousarray(ar)
     dtype = [('f{i}'.format(i=i), ar.dtype) for i in range(ar.shape[1])]
 
     try:
-        consolidated = ar.view(dtype)
+        if ar.shape[1] > 0:
+            consolidated = ar.view(dtype)
+        else:
+            consolidated = np.empty(len(ar), dtype=dtype)
     except TypeError:
```

Comment:
This `if` could probably do with a brief comment explaining why the special case is needed

Reply:
I pushed an update that added comments about this `if` statement.  Copy-editing suggestions are welcome!